scripts/dist-git/remove_unused_sources

```diff
@@ -0,0 +1,56 @@
+#!/usr/bin/bash
+
+Usage() {
+    cat <<EOF
+Usage:
+    $0 <package_git_directory> <package_lookaside_directory>
+
+    Removes all tarballs in <package_lookaside_directory> except
+    for tarballs that are referenced by the latest commit of each
+    branch in <package_git_directory>.
+EOF
+}
+
+if [[ $# != 2 ]]; then
+    Usage
+    exit 1
+fi
+
+pkg_git_dir="$1"
+pkg_lookaside_dir="$2"
+
+if [ ! -d "$pkg_git_dir" ]; then
+    echo "$pkg_git_dir is not a valid directory."
+    exit 1
+fi
+if [ ! -d "$pkg_lookaside_dir" ]; then
+    echo "$pkg_lookaside_dir is not a valid directory."
+    exit 1
+fi
+
+pushd $pkg_git_dir > /dev/null
+
+whitelist=()
+
+# find sources that are referenced by the latest commit in any of the branches
+for branch in `git for-each-ref --format="%(refname:short)" refs/heads`; do
+    while read line; do
+        set -- $line
+        filename=$2
+        test $# -gt 2 && filename=$(echo "$filename" | sed -e "s/^(//" -e "s/)$//")
+        whitelist+=($filename)
+    done < <(git show $branch:sources)
+done
+
+popd > /dev/null
+
+# remove all sources that are not referenced
+for source_dir in "$pkg_lookaside_dir"/*; do
+    srcdir="${source_dir##*/}"
+
+    if [[ "${whitelist[@]}" =~ "${srcdir}" ]]; then
```

Comment:
I have to admit this was new for me, but it doesn't seem to work as you expect?:
```
$ [[ "${a[@]}" =~ b ]] && echo yes
yes
```
I'd rather iterate over the whitelist: `for dir in "${whitelist[@]}"; do test "$dir" = "$srcdir"; ... ; done`.